library/core/src/primitive_docs.rs

```diff
@@ -1082,6 +1082,9 @@ impl<T> (T,) {}
 /// bits. Please see [the documentation for [`prim@f32`] or [Wikipedia on
 /// half-precision values][wikipedia] for more information.
 ///
+/// Note that not all major platforms have hardware support for f16, in which case a
+/// software implementation will be used. This may be slower than expected.
+///
```

Comment:
We can probably be more specific about what happens here, maybe:

```suggestion
/// Note that most major platforms will provide `f16` math support by converting to and from
/// an `f32`, which is usually fairly performant but will not be as fast as using `f32` directly.
```